client/repoutils.c

```diff
@@ -77,6 +176,12 @@ TDNFRepoGetBaseUrl(
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
+    if(!pRepos->pszBaseUrl)
```

Comment:
Is this check needed here, do we ever hit a NULL baseurl here? I believe this has to be checked when we retrieve the baseUrl from repo file or from metalink file right?

Reply:
We will not, added one condition in MDFileDownload.. Removed this.